concrete/src/StyleCustomizer/Style/ValueList.php

```diff
@@ -67,14 +67,27 @@ public static function loadFromRequest(ParameterBag $request, StyleList $styles)
         }
 
         if ($request->has('preset-fonts-file')) {
-            $bv = new BasicValue('preset-fonts-file');
-            $bv->setValue($request->get('preset-fonts-file'));
-            $vl->addValue($bv);
+            if (self::isValidFontPresetsFile($request->get('preset-fonts-file'))) {
+                $bv = new BasicValue('preset-fonts-file');
+                $bv->setValue($request->get('preset-fonts-file'));
+                $vl->addValue($bv);
+            }
         }
 
         return $vl;
     }
 
+    private static function isValidFontPresetsFile($path)
+    {
+        if (substr($path, -5) === '.less') {
+            $path = substr($path, 0, -5);
+            if (preg_match('/[A-Za-z0-9_-]/i', $path)) {
```

Comment:
Is this rx correct? Do we need to only check if $contains one of these characters?

Reply:
No – $path should only contain alpha-numeric, underscores and hyphens – no other characters allowed.

Reply:
So the rx should be
`'/^[A-Za-z0-9_-]+$/'`
right?

We could also use a single rx that comprises the previous check:

`'/^[A-Za-z0-9_-]+\.less$/'`

Or, simpler:

`'/^[\w-]+\.less$/'`

Reply:
Nice – I didn't even know about the `\w` feature. What do you think about the new regex in the updated version? I realized that I needed to include forward slashes in the allowed list of characters as well, so I've updated that as well. (We need to support things like `build/fonts/night-road.less`, etc...)

Reply:
Yep, that should be ok now.
Just one question: should we also check if $path doesn't start with a `/`? Can it be an absolute path?

Reply:
The definition of `\w` varies by locale. So, while this is simpler, it may have unintended consequences. 

I do agree that a positive (whitelist) approach of what _is_ allowed is easier to read and comprehend rather than a negative (blacklist) of finding anything that's not in the whitelist. The double negative requires some extra thought to process.

Unless we really want to support _any_ word character in _any_ locale, it's better to stick with character ranges (`[a-z0-9_-]`)

Reply:
@wordish are you sure that `\w` depends on the locale? AFAIK it only means US-ASCII letters (A-Z, a-z), digits (0-9) and underscore. And the letters are the 26 in the US-ASCII character set, without any diacritic mark.

Reply:
the `\w` character class doesn't depend on locale, it will always match `[A-Za-z0-9_]` this is true in all regex engines..
Although in php, java and python if you turn on the unicode modifier it will match `[\p{L}\p{N}_]`

examples of difference of \w modifier with and without unicode
**Without Unicode**
https://regex101.com/r/6PuKBr/1/
**With Unicode**
https://regex101.com/r/6PuKBr/2/

Though the simple regex `^[/\w-]+\.less$` catches a lot it does ask what is an acceptable filename?
is `/.less` acceptable? probably not  is `_.less` acceptable? or even `-.less`

If those aren't then you'd probably want to play around with the test cases on here using this regex `^[/\w-]*[\w]+\.less$`
https://regex101.com/r/6PuKBr/3/